File: solutions/70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1: return 1
        prev1 = 1
        prev2 = 2
        for i in range(2, n):
            cur = prev1 + prev2
            prev1 = prev2
            prev2 = cur
        return prev2

File: solutions/70-climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n < 3){
            return n;
        }

        int[] ans = new int[n];
        ans[0] = 1;
        ans[1] = 2;

        for(int i=3; i<=n; i++){
            ans[i-1] = ans[i-2] + ans[i-3];
        }

        return ans[n-1];
    }
}